_api-reference/msearch-template.md

```diff
@@ -0,0 +1,134 @@
+---
+layout: default
+title: Multi-search Template 
+nav_order: 47
+redirect_from: 
+ - /opensearch/rest-api/multi-search/
+---
+
+# Multi-search Template 
+
+**Introduced 1.0**
+{: .label .label-purple }
+
+The Multi-search Template API runs multiple search template requests in a single API request.
+
+## Path and HTTP methods
+
+The Multi-search template API uses the following paths:
+
+```
+GET /_msearch/template
+POST /_msearch/template
+GET /{index}/_msearch/template
+POST /{index}/_msearch/template
+```
+
+## Request body
+
+The multi-search template request body follows this pattern, similar to the [Multi-search API]({{site.url}}{{site.baseurl}}/api-reference/multi-search/):
+
+```
+Metadata\n
+Query\n
+Metadata\n
+Query\n
+
+```
+
+- Metadata lines include options, such as which indexes to search and the type of search.
+- Query lines use the [query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/).
+
+Like the [bulk]({{site.url}}{{site.baseurl}}/api-reference/document-apis/bulk/) operation, the JSON doesn't need to be minified---spaces are fine---but it does need to be on a single line. OpenSearch uses newline characters to parse multi-search requests and requires that the request body end with a newline character.
+
+## URL parameters and metadata options
+
+All multi-search template URL parameters are optional. Some can also be applied per-search as part of each metadata line.
+
+Parameter | Type | Description | Supported in metadata
+:--- | :--- | :---  | :---
+allow_no_indices | Boolean | Specifies whether to ignore wildcards that don't match any indexes. Default is `true`. | Yes
+cancel_after_time_interval | Time | The time after which the search request will be canceled. Supported at both parent and child request levels. The order of precedence is child-level parameter, parent-level parameter, and [cluster setting]({{site.url}}{{site.baseurl}}/api-reference/cluster-settings/). Default is `-1`. | Yes
+css_minimize_roundtrips | Boolean | Specifies whether OpenSearch should try to minimize the number of network round trips between the coordinating node and remote clusters (only applicable to cross-cluster search requests). Default is `true`. | No
+expand_wildcards | Enum | Expands wildcard expressions to concrete indexes. Combine multiple values with commas. Supported values are `all`, `open`, `closed`, `hidden`, and `none`. Default is `open`. | Yes
+ignore_unavailable | Boolean | If an index or shard from the indexes list doesn’t exist, then this setting specifies whether to ignore the missing index or shard rather than fail the query. Default is `false`. | Yes
```

Comment:
```suggestion
ignore_unavailable | Boolean | If an index or shard from the index list doesn’t exist, then this setting specifies whether to ignore the missing index or shard rather than fail the query. Default is `false`. | Yes
```